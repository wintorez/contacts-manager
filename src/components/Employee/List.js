import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Page from '../Page';
import { Table, Button, Loader, Message } from 'semantic-ui-react';

const { Header, Row, HeaderCell, Cell, Body } = Table;

const List = ({ location, history, loading, error, employees }) => (
  <Page title="Employees" location={location}>
    {loading && <Loader active inline="centered" />}
    {error && (
      <Message negative>
        Oops! Something went wrong! Please try again later.
      </Message>
    )}
    {!error &&
      !loading && (
        <Fragment>
          <Table celled selectable>
            <Header>
              <Row>
                <HeaderCell>Employee Id</HeaderCell>
                <HeaderCell>First Name</HeaderCell>
                <HeaderCell>Last Name</HeaderCell>
                <HeaderCell>Email</HeaderCell>
              </Row>
            </Header>
            <Body>
              {employees.map(({ employeeId, firstName, lastName, email }) => (
                <Row
                  key={employeeId}
                  style={{ cursor: 'pointer' }}
                  onClick={() => history.push(`/employees/${employeeId}`)}
                >
                  <Cell>{employeeId}</Cell>
                  <Cell>{firstName}</Cell>
                  <Cell>{lastName}</Cell>
                  <Cell>{email}</Cell>
                </Row>
              ))}
            </Body>
          </Table>
          <Button onClick={() => history.push('/employees/add')}>
            Add New Employee
          </Button>
        </Fragment>
      )}
  </Page>
);

List.propTypes = {
  location: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  employees: PropTypes.array.isRequired
};

export default List;
