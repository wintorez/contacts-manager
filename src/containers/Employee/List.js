import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import List from '../../components/Employee/List';

const mapStateToProps = (state, ownProps) => ({
  loading: state.status.loading,
  error: state.status.error,
  employees: state.employees
});

const mapDispatchToProps = (dispatch, ownProps) => ({});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(List)
);
