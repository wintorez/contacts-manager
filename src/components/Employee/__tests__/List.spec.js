import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import List from '../List';

const defaults = {
  history: {},
  updateEmployee: jest.fn(),
  employees: [
    {
      employeeId: '1',
      firstName: 'Joe',
      lastName: 'Bloggs',
      email: 'jblogs@email.com',
      telephone: '+1-555-444-3333',
      department: 'Engineering'
    },
    {
      employeeId: '2',
      firstName: 'John',
      lastName: 'Smith',
      email: 'jsmith@gmail.com',
      telephone: '+1-777-888-9999',
      department: 'Marketing & Sales'
    }
  ]
};

describe('<List />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <List {...defaults} location={{ pathname: '/employees' }} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <List {...defaults} location={{ pathname: '/employees' }} loading />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when error', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <List {...defaults} location={{ pathname: '/employees' }} error />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
