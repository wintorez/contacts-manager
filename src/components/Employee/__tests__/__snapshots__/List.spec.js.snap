// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders correctly 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui container"
  >
    <table
      className="ui celled selectable table"
    >
      <thead
        className=""
      >
        <tr
          className=""
        >
          <th
            className=""
          >
            Employee Id
          </th>
          <th
            className=""
          >
            First Name
          </th>
          <th
            className=""
          >
            Last Name
          </th>
          <th
            className=""
          >
            Email
          </th>
        </tr>
      </thead>
      <tbody
        className=""
      >
        <tr
          className=""
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <td
            className=""
          >
            1
          </td>
          <td
            className=""
          >
            Joe
          </td>
          <td
            className=""
          >
            Bloggs
          </td>
          <td
            className=""
          >
            jblogs@email.com
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <td
            className=""
          >
            2
          </td>
          <td
            className=""
          >
            John
          </td>
          <td
            className=""
          >
            Smith
          </td>
          <td
            className=""
          >
            jsmith@gmail.com
          </td>
        </tr>
      </tbody>
    </table>
    <button
      className="ui button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Add New Employee
    </button>
  </div>
</div>
`;

exports[`<List /> renders correctly when error 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui container"
  >
    <div
      className="ui negative message"
    >
      Oops! Something went wrong! Please try again later.
    </div>
  </div>
</div>
`;

exports[`<List /> renders correctly when loading 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui container"
  >
    <div
      className="ui active centered inline loader"
    />
  </div>
</div>
`;
